config/fi_provider.m4

```diff
@@ -192,3 +193,56 @@ AC_DEFUN([_FI_PROVIDER_INVOKE],[
 		[])]
 	    )
 ])
+
+dnl
+dnl Internal; should not be called from provder .m4 scripts.
+dnl Helper macro that appends each providers build flags for pkg-config
+dnl
+dnl Arguments:
+dnl
+dnl $1: name of the provider
+dnl
+AC_DEFUN([_FI_PROVIDER_PC_ADD],[
+	 FI_PC_CFLAGS="$FI_PC_CFLAGS $$1_CFLAGS $$1_CPPFLAGS"
+	 FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_LIBS"
+	 # usnic with nl needs special handling (should be added to usnic_LIBS?)
+	 AS_IF([test "$1" == "usnic"], [
```

Comment:
Yes, I didn't want to muck with the provider-specific macros until getting feedback.  I'd say either have each provider fill in the PC vars as necessary or try and condense to $$1_LIBS.